### PR TITLE
Bug 1940059: Add ceph-dashboard link for OCS external cluster overview page

### PR DESCRIPTION
**Pull request by TimothyAsirJeyasing**
Read ceph-dashboard url from the secret and add the link
for ocs external cluster overview page under storage tab in UI.
image
image

Signed-off-by: Timothy Asir Jeyasingh <tjeyasin@redhat.com>

### CODE REVIEW

**Comment by TimothyAsirJeyasing** [comment]
@bipuladh Please review

File: frontend/packages/ceph-storage-plugin/src/components/dashboards/persistent-external/details-card.tsx
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/src/components/dashboards/persistent-external/details-card.tsx:112]:
```suggestion
```

File: frontend/packages/ceph-storage-plugin/src/components/dashboards/persistent-external/details-card.tsx
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/src/components/dashboards/persistent-external/details-card.tsx:114]:
```suggestion
```

**Review by bipuladh — CHANGES_REQUESTED** [review]

File: frontend/packages/ceph-storage-plugin/src/components/dashboards/persistent-external/details-card.tsx
Comment (SanjalKatiyar) [inline frontend/packages/ceph-storage-plugin/src/components/dashboards/persistent-external/details-card.tsx:21]: We are already passing `watchK8sResource` & `stopWatchK8sResource` to the card as a prop. We can use that directly. We don't have to import these hooks here, right ?
^^ @bipuladh

Comment (bipuladh) [reply frontend/packages/ceph-storage-plugin/src/components/dashboards/persistent-external/details-card.tsx:21]: agreed

**Review by bipuladh — APPROVED** [review]
/lgtm

File: frontend/packages/ceph-storage-plugin/src/components/dashboards/persistent-external/details-card.tsx
Comment (SanjalKatiyar) [inline frontend/packages/ceph-storage-plugin/src/components/dashboards/persistent-external/details-card.tsx]: Sorry, I missed this earlier,
instead of directly passing `false`... we can use:
```
isLoading = !cephSecret?.loaded;
error = cephSecret?.loadError;
```

**Comment by cloudbehl** [comment]
/hold
Till UX reviews it. @yuvalgalanti

**Comment by cloudbehl** [comment]
/retitle Bug 1940059: Add ceph-dashboard link for OCS external cluster overview page

**Comment by yuvalgalanti** [comment]
looks good @TimothyAsirJeyasing

**Comment by bipuladh** [comment]
@yuvalgalanti Is Ceph Dashboard a term we want to expose here? Should it perhaps be RedHat Ceph Storage Dashboard?

**Comment by bipuladh** [comment]
@TimothyAsirJeyasing please change the Item name to `RedHat Ceph Storage Dashboard`?

**Comment by bipuladh** [comment]
/hold
testing

**Review by SanjalKatiyar — APPROVED** [review]

**Comment by SanjalKatiyar** [comment]
tested working fine, link is accessible when `rook-ceph-dashboard-link` secret is present otherwise link is hidden.
/hold cancel

**Comment by SanjalKatiyar** [comment]
looks good will tag it once tests are working fine.

**Comment by yuvalgalanti** [comment]
looks very good @TimothyAsirJeyasing

File: frontend/packages/ceph-storage-plugin/src/components/dashboards/persistent-external/details-card.tsx
Comment (afreen23) [reply frontend/packages/ceph-storage-plugin/src/components/dashboards/persistent-external/details-card.tsx:110]: Define `CEPH_BRAND_NAME` in `packages/ceph-storage-plugin/src/constants/common.ts`

**Review by afreen23 — CHANGES_REQUESTED** [review]

**Comment by afreen23** [comment]
/lgtm cancel
Some reviews required https://github.com/openshift/console/pull/9398#pullrequestreview-725094733|
cc @cloudbehl

File: frontend/packages/ceph-storage-plugin/src/components/dashboards/persistent-external/details-card.tsx
Comment (SanjalKatiyar) [reply frontend/packages/ceph-storage-plugin/src/components/dashboards/persistent-external/details-card.tsx:79]: Not using `useDeepCompareMemoize` is better, I agree...
One suggestion, it is better to use `useCallback` for `getCephLink` fucntion so that we do not Base64.decode the secret again if some other resource (other than secret) changes in the current FC... wdyt, @afreen23 ??

Comment (afreen23) [reply frontend/packages/ceph-storage-plugin/src/components/dashboards/persistent-external/details-card.tsx:79]: Its okay until we see any performance issues.

Comment (SanjalKatiyar) [reply frontend/packages/ceph-storage-plugin/src/components/dashboards/persistent-external/details-card.tsx:79]: What I meant that, we are watching for 3 resources total in that particular FC... `OCSServiceModel`, `SubscriptionModel` & `SecretModel` ...
So, if `SecretModel` does not change, but, any of `OCSServiceModel`, `SubscriptionModel` changes then also it will decode the sercet again.

But, yeah this also make sense --> Its okay until we see any performance issues.

File: frontend/packages/ceph-storage-plugin/src/components/dashboards/persistent-external/details-card.tsx
Comment (SanjalKatiyar) [reply frontend/packages/ceph-storage-plugin/src/components/dashboards/persistent-external/details-card.tsx:110]: @afreen23 I more issue if we do this way:
```
          <DetailItem
            title={t('ceph-storage-plugin~Provider')}
            isLoading={!secretLoaded}
            error={!!secretError}
          >
           {link ?  <ExternalLink href={link} text={CEPH_BRAND_NAME}} /> :    CEPH_BRAND_NAME
          </DetailItem>
        )}
```
if `link` is `null` it means there is some error retrieving the secret  (say NotFound error), in that case  `DetailItem` will by default display `Not Available` instead of CEPH_BRAND_NAME (.
So, either Tim need to do as he was doing before (two `DetailItem` on basis on ternary condition) or we pass `error={false} && isLoading={false}` always to `DetailItem`.

Comment (afreen23) [reply frontend/packages/ceph-storage-plugin/src/components/dashboards/persistent-external/details-card.tsx:110]: I synced on a call with Tim, the behavior is few customers can and cannot have ceph dashboard.

File: frontend/packages/ceph-storage-plugin/src/components/dashboards/persistent-external/details-card.tsx
Comment (SanjalKatiyar) [inline frontend/packages/ceph-storage-plugin/src/components/dashboards/persistent-external/details-card.tsx:91]: Is this a typo ?

File: frontend/packages/ceph-storage-plugin/src/components/dashboards/persistent-external/details-card.tsx
Comment (SanjalKatiyar) [inline frontend/packages/ceph-storage-plugin/src/components/dashboards/persistent-external/details-card.tsx:97]: Do we not want to externalise it anymore ?

File: frontend/packages/ceph-storage-plugin/src/components/dashboards/persistent-external/details-card.tsx
Comment (TimothyAsirJeyasing) [reply frontend/packages/ceph-storage-plugin/src/components/dashboards/persistent-external/details-card.tsx:91]: yes will remove now

File: frontend/packages/ceph-storage-plugin/src/components/dashboards/persistent-external/details-card.tsx
Comment (SanjalKatiyar) [reply frontend/packages/ceph-storage-plugin/src/components/dashboards/persistent-external/details-card.tsx:97]: {cephLink ? <ExternalLink href={cephLink} text={CEPH_BRAND_NAME} /> : cephBrandName(t)}

File: frontend/packages/ceph-storage-plugin/src/constants/common.ts
Comment (SanjalKatiyar) [inline frontend/packages/ceph-storage-plugin/src/constants/common.ts:5]: `export const cephBrandName = (t: TFunction) => t('ceph-storage-plugin~Red Hat Ceph Storage');`
also, `import { TFunction } from 'i18next';` on the top.

File: frontend/packages/ceph-storage-plugin/src/components/dashboards/persistent-external/details-card.tsx
Comment (SanjalKatiyar) [inline frontend/packages/ceph-storage-plugin/src/components/dashboards/persistent-external/details-card.tsx:20]: import { CEPH_STORAGE_NAMESPACE, cephBrandName } from '../../../constants';

File: frontend/packages/ceph-storage-plugin/src/components/dashboards/persistent-external/details-card.tsx
Comment (afreen23) [reply frontend/packages/ceph-storage-plugin/src/components/dashboards/persistent-external/details-card.tsx:97]: Its a product name, we can ignore that.

Comment (SanjalKatiyar) [reply frontend/packages/ceph-storage-plugin/src/components/dashboards/persistent-external/details-card.tsx:97]: ack !!

**Review by afreen23 — APPROVED** [review]
/lgtm
I am seeing @SanjalKatiyar already approved and no pending reviews.
The changes are good now.

**Pull request merged**
